core/color.js

```diff
@@ -0,0 +1,432 @@
+/**
+ * @license Copyright (c) 2003-2020, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+/**
+ * @fileoverview Defines the {@link CKEDITOR.tools.style.color} normalizer class
+ * that parse color string code other formats.
+ */
+
+( function() {
+	'use strict';
+
+	CKEDITOR.tools.style.color = function( colorCode ) {
```

Comment:
I thought I have mentioned it earlier but couldn't find it :thinking: I thing it shouldn't be in `style` namespace since it's more generic. So I would be fro moving it to `CKEDITOR.tools.color` namespace.

So all my previous comments mentioning `CKEDITOR.tools.style.color` means `CKEDITOR.tools.color` color namespace too :see\_no\_evil: :smile:

Reply:
And since we get rid of circular dependency we can get back to `createClass` (as it is also good practice mentioned in our code style guide - https://github.com/ckeditor/ckeditor4/blob/major/dev/docs/codestyle.md#2226-constructor-declaration).

Sorry for the confusion since I mentioned earlier to get rid of it :thinking: :speak\_no\_evil:

Reply:
Originally you said it should be in `CKEDITOR.tools.color`. It was on slack chat I think. Then `style` namespace was suggested by @jacekbogdanski in another private chat.

Anyway - I move it to `CKEDITOR.tools.color`